File: server/bussines/empresa.js
const empresaDao = require("../dao/empresaDao.js");

const crearEmpresa = function(req, res){
    console.log(req.body.rut);
    let empresa = 
    {
        rut : parseInt(req.body.rut),
        nombre : req.body.nombre,
        direccion : req.body.direccion,
        telefono : req.body.telefono
    };
    empresaDao.crearEmpresa(empresa).then((success) => {
        console.log(success.toString());
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};

const actualizarEmpresa = function(req, res){
    let empresa = 
    {
        rut : parseInt(req.body.rut),
        nombre : req.body.nombre,
        direccion : req.body.direccion,
        telefono : req.body.telefono
    };
    empresaDao.actualizarEmpresa(empresa).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};


const actualizarEstadoEmpresa = function(req, res){
    let empresa = 
    {
        rut : req.body.rut,
        estado : req.body.estado
    };
    empresaDao.actualizarEstadoEmpresa(empresa).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};

const obtenerEmpresaPorId = function(req, res){
    let empresa = 
    {
        rut : req.body.rut
    };
    empresaDao.obtenerEmpresaPorId(empresa).then((success) => {
        res.send(success);
    }).catch((error) => {
        console.log(error);
    });
};

const obtenerEmpresas = function(req, res){
    empresaDao.obtenerEmpresas().then((success) => {
        res.send(success);
    }).catch((error) => {
        console.log(error);
    });
};

module.exports.crearEmpresa = crearEmpresa;
module.exports.actualizarEmpresa = actualizarEmpresa;
module.exports.actualizarEstadoEmpresa = actualizarEstadoEmpresa;
module.exports.obtenerEmpresas = obtenerEmpresas;
module.exports.obtenerEmpresaPorId = obtenerEmpresaPorId;

File: Desktop/Desktop.Data/Empresa.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace Desktop.Data
{
    public class Empresa
    {
        private string _url;
        public int Rut { get; set; }
        public string Nombre { get; set; }
        public string Direccion { get; set; }
        public string Telefono { get; set; }
        public bool Estado { get; set; }

        public Empresa()
        {
            _url = "http://127.0.0.1:3000/";
        }

        public bool CrearEmpresa(Empresa empresa)
        {
            Conexion.Cliente.BaseAddress = new Uri(_url);
            Conexion.Url = _url;
            var queryParams = new Dictionary<string, string>
            {
                { "rut", empresa.Rut.ToString() },
                { "nombre", empresa.Nombre },
                { "direccion", empresa.Direccion },
                { "telefono", empresa.Telefono.ToString() },
            };
            return Conexion.Post(queryParams, "crearEmpresa"); ;
        }

        public bool ActualizarEmpresa(Empresa empresa)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "rut", empresa.Rut.ToString() },
                { "nombre", empresa.Nombre },
                { "direccion", empresa.Direccion },
                { "telefono", empresa.Telefono.ToString() },
            };
            return Conexion.Post(queryParams, "actualizarEmpresa");
        }

        public bool ActualizarEstadoEmpresa(int rut, bool estado)
        {
            Conexion.Url = _url;

            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "rut", rut.ToString() },
                { "estado", estado ? "1" : "0" },
            };
            return Conexion.Post(queryParams, "actualizarEstadoEmpresa");
        }

        public List<Empresa> ObtenerEmpresas()
        {
            Conexion.Url = _url;

            Conexion.Cliente.BaseAddress = new Uri(_url);
            var res = Conexion.Cliente.GetStringAsync(_url + "obtenerEmpresas").Result;
            List<Empresa> empresas = JsonConvert.DeserializeObject<List<Empresa>>(res);
            return empresas;
        }

        public Empresa ObtenerEmpresa(int rut)
        {
            Conexion.Url = _url;

            var url = new UriBuilder(_url);
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string> { { "rut", rut.ToString() } };
            var content = new FormUrlEncodedContent(queryParams);
            var res = Conexion.Cliente.PostAsync(_url + "obtenerEmpresaPorId/?rut=", content).Result
                .Content.ReadAsStringAsync().Result;
            Empresa empresa = JsonConvert.DeserializeObject<List<Empresa>>(res)[0];
            return empresa;
        }
    }
}
